client/transaction-pool/src/api.rs

```diff
@@ -136,45 +139,66 @@ impl<Client, Block> sc_transaction_graph::ChainApi for FullChainApi<Client, Bloc
 		self.client.to_hash(at).map_err(|e| Error::BlockIdConversion(format!("{:?}", e)))
 	}
 
-	fn hash_and_length(&self, ex: &sc_transaction_graph::ExtrinsicFor<Self>) -> (Self::Hash, usize) {
+	fn hash_and_length(
+		&self,
+		ex: &sc_transaction_graph::ExtrinsicFor<Self>,
+	) -> (Block::Hash, usize) {
 		ex.using_encoded(|x| {
 			(<traits::HashFor::<Block> as traits::Hash>::hash(x), x.len())
 		})
 	}
+
+	fn leaves(&self) -> Result<Vec<Block::Hash>, Self::Error> {
+		self.backend.blockchain().leaves().map_err(Into::into)
+	}
+
+	fn tree_route(
```

Comment:
I think this is no longer needed, since `tree_route` is part of the notification?

Reply:
It is, the `tree_route` that is part of the notification is only the one from the re-org, however there can be multiple leaves as described in the pr. However, I can merge both functions.